module/VuFind/src/VuFind/Auth/Shibboleth.php

```diff
@@ -228,18 +274,14 @@ public function getSessionInitiator($target)
      */
     public function isExpired()
     {
-        $config = $this->getConfig();
-        if (isset($config->Shibboleth->username)
-            && isset($config->Shibboleth->logout)
-        ) {
-            // It would be more proper to call getServer on a Laminas request
-            // object... except that the request object doesn't exist yet when
-            // this routine gets called.
-            $username = isset($_SERVER[$config->Shibboleth->username])
-                ? $_SERVER[$config->Shibboleth->username] : null;
-            return empty($username);
-        }
-        return false;
+        // It would be more proper to call getServer on a Laminas request
```

Comment:
These changes drew my attention on this method and the logout mechanism. I can see it's ancient, but it looks like it might be something that might cause trouble too. E.g. in our case we only enable Shibboleth in Apache when the url contains `auth_method=Shibboleth`, and even then the actual Shibboleth session is short-lived. I suppose we haven't stumbled on this since we use ChoiceAuth and it doesn't implement isExpired(). So, would you be able to explain the use case or whether this is just some ancient code no longer really needed? If this is still necessary for a certain case, can you check the comment and add a configuration setting that explicitly enables the expiration check? It was previously tied to config containing username and logout settings, but e.g. adding the logout setting enabling this would have seem like a suprising side effect.